doc/website/blog/2022-03-31-ent-hooks-examples.md

```diff
@@ -0,0 +1,411 @@
+---
+title: How I use Ent Schema Hooks
+author: Yoni Davidson
+authorURL: "https://github.com/yonidavidson"
+authorImageURL: "https://avatars0.githubusercontent.com/u/5472778"
+authorTwitter: yonidavidson
+---
+Despite being one of the most powerful features of Ent, Schema [hooks](https://entgo.io/docs/hooks) 
+are often overlooked by many users. We've covered hooks in previous blog posts (such as 
+[Building Observable Ent Applications with Prometheus](/blog/2021/08/12/building-observable-ent-application-with-prometheus)
+and [Sync Changes to External Data Systems using Ent Hooks](/blog/2021/11/1/sync-to-external-data-systems-using-hooks)),
+but learning from my personal experience building real applications with Ent, I thought it would be
+beneficial to put them in the spotlight once more.
+
+### What are hooks?
+
+[Hooks](https://entgo.io/docs/hooks) are an Ent feature that allow adding custom logic before and after operations that change the data entities.
+
+Hooks are functions that get and return an [ent.Mutator](https://pkg.go.dev/entgo.io/ent#Mutator).
+They function similar to the popular HTTP middleware pattern.
+
+```go
+package example
+
+import (
+	"context"
+
+	"entgo.io/ent"
+)
+
+func exampleHook() ent.Hook {
+	return func(next ent.Mutator) ent.Mutator {
+		return ent.MutateFunc(func(ctx context.Context, m ent.Mutation) (ent.Value, error) {
+			// Do something before mutation.
+			v, err := next.Mutate(ctx, m)
+			if err != nil {
+				// Do something if error after mutation.
+			}
+			// Do something after mutation.
+			return v, err
+		})
+	}
+}
+```
+
+[Schema hooks](https://entgo.io/docs/hooks#schema-hooks) are mainly used for defining custom mutation logic on a specific entity type, for example,
+syncing entity creation to another system.
+
+### Schema as code
+
+In Ent, we try and push the entity's logic to the schema level, this way, the logic can be near the schema and not located
+in different places in a system. Entity properties could be:
+1. Field type (Text, Int, Boolean)
+2. Field Valid values ("Not empty","positive")
+3. Immutability status for each field
+4. Privacy
+5. Required Side effects (heavy compute for example)
```

Comment:
Something with the flow of the sentence does not work for me ("Entity properties could be..."). 

Maybe @hilakashai can help with this.